src/internal/components/selectable-item/index.tsx

```diff
@@ -3,11 +3,63 @@
 import React, { useLayoutEffect, useRef } from 'react';
 import clsx from 'clsx';
 
+import { getAnalyticsMetadataAttribute } from '@cloudscape-design/component-toolkit/internal/analytics-metadata';
+
 import { BaseComponentProps, getBaseProps } from '../../base-component';
 import { HighlightType } from '../options-list/utils/use-highlight-option.js';
+import { GeneratedAnalyticsMetadataSelectableItemSelect } from './analytics-metadata/interfaces';
 
+import optionAnalyticsSelectors from './../option/analytics-metadata/styles.css.js';
+import analyticsSelectors from './analytics-metadata/styles.css.js';
 import styles from './styles.css.js';
 
+export interface ItemDataAttributes {
+  'data-group-index'?: string;
+  'data-child-index'?: string;
+  'data-in-group-index'?: string;
+  'data-test-index'?: string;
```

Comment:
These data attributes only exists inside the  autosuggest [option](https://github.com/cloudscape-design/components/blob/09aa16f2933bcd5cf004dad5f5c82cedb5a93251/src/autosuggest/autosuggest-option.tsx#L83-L85) `<SelectableItem/>` component, the other references does not have any data attributes, is this expected?
- [Item](https://github.com/cloudscape-design/components/blob/09aa16f2933bcd5cf004dad5f5c82cedb5a93251/src/select/parts/item.tsx#L65)
- [Multiselect](https://github.com/cloudscape-design/components/blob/09aa16f2933bcd5cf004dad5f5c82cedb5a93251/src/select/parts/multiselect-item.tsx#L56)

Reply:
They appear in many places:
- https://github.com/search?q=repo%3Acloudscape-design%2Fcomponents%20data-in-group-index&type=code
- https://github.com/search?q=repo%3Acloudscape-design%2Fcomponents%20data-group-index&type=code
- https://github.com/search?q=repo%3Acloudscape-design%2Fcomponents%20data-child-index&type=code

It is actually inconsistent, because sometimes we use `data-child-index` and `data-in-group-index` for the same purpose. However, I didn't want to refactor those attributes because we use them in test utilities, so there might be broken builds.